api/src/utils/sql.test.ts

```diff
@@ -0,0 +1,296 @@
+      });
+    });
+  });
+
+  describe('objectToCamelCase', () => {
+    it('should convert object keys from snake_case to camelCase', () => {
+      const input = {
+        product_id: 1,
+        supplier_id: 2,
+        unit_price: 10.99,
+      };
+      const output = objectToCamelCase(input);
+      expect(output).toEqual({
+        productId: 1,
+        supplierId: 2,
+        unitPrice: 10.99,
+      });
+    });
+  });
+
+  describe('mapDatabaseRows', () => {
+    it('should convert multiple database rows to typed models', () => {
+      const rows = [
+        { product_id: 1, product_name: 'Product 1', unit_price: 10.99 },
+        { product_id: 2, product_name: 'Product 2', unit_price: 20.99 },
+      ];
+      const mapped = mapDatabaseRows(rows);
+      expect(mapped).toEqual([
+        { productId: 1, productName: 'Product 1', unitPrice: 10.99 },
+        { productId: 2, productName: 'Product 2', unitPrice: 20.99 },
+      ]);
+    });
+
+    it('should handle empty array', () => {
+      const rows: any[] = [];
+      const mapped = mapDatabaseRows(rows);
+      expect(mapped).toEqual([]);
+    });
+  });
+
+  describe('generatePlaceholders', () => {
+    it('should generate correct number of placeholders', () => {
+      expect(generatePlaceholders(1)).toBe('?');
+      expect(generatePlaceholders(3)).toBe('?, ?, ?');
+      expect(generatePlaceholders(5)).toBe('?, ?, ?, ?, ?');
+    });
+
+    it('should handle zero placeholders', () => {
+      expect(generatePlaceholders(0)).toBe('');
+    });
+  });
+
+  describe('buildInsertSQL', () => {
+    it('should build correct INSERT SQL with values', () => {
+      const data = {
+        name: 'Test Product',
+        price: 99.99,
+        supplierId: 1,
+      };
+      const { sql, values } = buildInsertSQL('products', data);
+      
+      expect(sql).toBe('INSERT INTO products (name, price, supplier_id) VALUES (?, ?, ?)');
+      expect(values).toEqual(['Test Product', 99.99, 1]);
+    });
+
+    it('should convert boolean values to integers', () => {
+      const data = {
+        active: true,
+        verified: false,
+      };
+      const { sql, values } = buildInsertSQL('suppliers', data);
+      
+      expect(values).toEqual([1, 0]); // true becomes 1, false becomes 0
+    });
+  });
+
+  describe('buildUpdateSQL', () => {
+    it('should build correct UPDATE SQL with WHERE clause', () => {
+      const data = {
+        name: 'Updated Name',
+        price: 79.99,
+      };
+      const { sql, values } = buildUpdateSQL('products', data, 'product_id = ?');
+      
+      expect(sql).toBe('UPDATE products SET name = ?, price = ? WHERE product_id = ?');
+      expect(values).toEqual(['Updated Name', 79.99]);
+    });
+
+    it('should convert boolean values to integers in updates', () => {
+      const data = {
+        active: false,
+        verified: true,
+      };
+      const { sql, values } = buildUpdateSQL('suppliers', data, 'supplier_id = ?');
```

Comment:
Unused variable sql.
```suggestion
      const { values } = buildUpdateSQL('suppliers', data, 'supplier_id = ?');
```